### ISSUE TITLE
Resolve/Include file overrides for Parent Child theming

### ISSUE THREAD

**Issue by kevinsperrine** [vuejs/vue-loader#686]
I have a parent->child theme where if the file exists in the child, I want it to load and ignore the parent. However, if the file does not exist in the child the parent file should be loaded. I've modified my webpack configs to resolve modules from the child first and then the parent second and adjusted all the loader includes to load from the child also (eslint, ts-loader, css-loaders, sass-loaders, etc). The styles are loaded from the vue file, the templates are loaded within a typescript file, and the scripts are imported from the same typescript file. Everything works great _except_ when overloading the components typescript file and leaving the original vue file in the parent. In this situation the styles get dropped completely; neither the parent nor the child styles for the component are extracted in to the css. I know this isn't a normal setup, but I wanted to see what is causing the styles to be dropped. Is the overloading of the Vue component class causing the original vue file to not output styles for some reason?

```
// folder structure
+ parent/src/components/header/{Header.vue,Header.ts,header-template.ts,_header.scss}

// template is overridden correctly and styled as parent.
+ child/src/components/header/header-template.ts

// template is overridden correctly and styled as child.
+ child/src/components/header/{header-template.ts,_header.scss}

// template is overridden correctly, scripts are overridden correctly, but styles are dropped entirely.
+ child/src/components/header/{header-template.ts,_header.scss,Header.ts}

// template is overridden correctly, scripts are overridden correctly, but styles are dropped entirely (no parent or child)
+ child/src/components/header/{header-template.ts,_header.scss,Header.ts,Header.vue}

// fails entirely Module build failed: Error: Could not find file: 
// (even though the given path is exactly correct to the location of the child vue file). 
// I'm assuming this is a configuration error somewhere on my part, but haven't tracked it down yet. 
// I've had to duplicate tsconfig.json, .eslintrc.js, and install all node_modules in the child directory, too.
// I assume this is related and somehow caused by relative paths in a config, but I'm not sure.
+ child/src/components/header/Header.vue
```

```
// parent/src/components/header/Header.vue
<script lang="ts">
import Header from 'components/header/Header.ts'
export default Header
</script>
<style lang="scss">
  @import "~components/header/header";
</style>
```

```
// parent/src/components/header/Header.ts
import Vue from 'vue'
import Component from 'vue-class-component'
import { template } from 'components/header/header-template'

@Component({
    name: "Header",
    template: template,
    data () {
        return {
            title: 'Parent'
        }
    },
})
export default class Header extends Vue {

}
```

```
// parent/src/components/header/header-template.ts
export const template: string = `
  <header>
    <h1>Header</h1>
  </header>
`
```

```
// parent/src/components/header/_header.scss
header {
    background-color: #000;
    color: #fff;
}
```

```
// webpack config
  resolve: {
    extensions: ['.js', '.vue', '.json', '.ts'],
    modules: [
      resolve('../../child/app/src'),
      resolve('src'),
      "node_modules"
    ],
...
  module: {
    rules: [
      {
        test: /\.(js|vue)$/,
        loader: 'eslint-loader',
        enforce: "pre",
        include: [
          resolve('../../child/app/src'),
          resolve('src'),
          resolve('test')
        ],
...
      {
        test: /\.ts$/,
        loader: 'ts-loader',
        include: [
          resolve('../../child/app/src'),
          resolve('src'),
          resolve('test')
        ],
```

```
// vue-loader.conf
module.exports = {
  loaders: utils.cssLoaders({
    sourceMap: isProduction
      ? config.build.productionSourceMap
      : config.dev.cssSourceMap,
    extract: isProduction
  }),
  include: [
    resolve('../../child/app/src'),
    resolve('src')
  ],
  esModule: true
}
```

```
// css loaders
module.exports = {
  loaders: utils.cssLoaders({
    sourceMap: isProduction
      ? config.build.productionSourceMap
      : config.dev.cssSourceMap,
    extract: isProduction
  }),
  include: [
    resolve('../../child/app/src'),
    resolve('src')
  ],
  esModule: true
}
```

**Comment by LinusBorg** [comment]
This is quite hard to follow without the ability to acutally run the code. Can you provide repository with a minimum reproduction?

**Comment by sodatea** [comment]
Closing due to inactivity. Please open a new issue with a reference to this one if you can follow up with more information.
